### PR TITLE
Audit To-Hit for Polearms

### PR DESCRIPTION
**Pull request by ANickelN**
Most polearms were changed to have "neutral" balance, because while they may be long with a weight on one end, they are MEANT to be used for stabbing so it didn't seem appropriate to give them the "uneven" tag given that the description of the "good" balance tag says "Well-balanced for swinging or stabbing."

Makeshift or crude spears or polearms were given the "solid" grip rather than the "weapon" grip attribute.

#### Summary
Balance "Audit To-Hit for Polearms"

#### Purpose of change

To adjust the to-hit values of polearms to match game_balance.md

#### Describe the solution

Go through the files and correct values according to game_balance.md

#### Describe alternatives you've considered

Leaving things as they are.

#### Testing

Spawned items in-game, to-hit values were changed properly.

### CODE REVIEW

File: tests/effective_dps_test.cpp [hunk 1]
```diff
@@ -286,41 +286,41 @@ static void check_staves( const std::function<Approx( const std::string & )> &ca
...
+        CHECK( calc_expected_dps( "pitchfork" ) == 12.48 );
+        CHECK( calc_expected_dps( "spear_stone" ) == 10.42 );
+        CHECK( calc_expected_dps( "spear_forked" ) == 12.15 );
+        CHECK( calc_expected_dps( "pike_fake" ) == 7.55 );
     }
 }
 
 static void check_polearms( const std::function<Approx( const std::string & )> &calc_expected_dps )
 {
     SECTION( "polearms" ) { // typical value around 35
```
Comment (anothersimulacrum) [inline tests/effective_dps_test.cpp:309]: I'd stay at most a point under the typical value, I recommend dropping all the ones that popped up into the 40s down into the 35-36 range.

File: tests/effective_dps_test.cpp [hunk 2]
```diff
@@ -286,41 +286,41 @@ static void check_staves( const std::function<Approx( const std::string & )> &ca
 static void check_spears( const std::function<Approx( const std::string & )> &calc_expected_dps )
 {
     SECTION( "spears" ) { // typical value around 24
-        CHECK( calc_expected_dps( "spear_steel" ) == 24.5 );
+        CHECK( calc_expected_dps( "spear_steel" ) == 23.19 );
         CHECK( calc_expected_dps( "pike" ) == 23.0 );
-        CHECK( calc_expected_dps( "qiang" ) == 23.0 );
-        CHECK( calc_expected_dps( "spear_dory" ) == 23 );
-        CHECK( calc_expected_dps( "spear_homemade_halfpike" ) == 20.0 );
+        CHECK( calc_expected_dps( "qiang" ) == 26.54 );
```
Comment (anothersimulacrum) [inline tests/effective_dps_test.cpp:291]: This should get a drop, it's well above the others in its category.

File: tests/effective_dps_test.cpp (hunk 1 above)
Comment (ANickelN) [reply tests/effective_dps_test.cpp:309]: Okay, nerfed them and we'll see if they get lowered enough.

File: data/json/items/melee/spears_and_polearms.json [hunk 3]
```diff
@@ -547,7 +547,7 @@
     "longest_side": "180 cm",
     "bashing": 6,
     "cutting": 40,
-    "to_hit": { "grip": "weapon", "length": "long", "surface": "point", "balance": "uneven" },
+    "to_hit": { "grip": "weapon", "length": "long", "surface": "line", "balance": "neutral" },
```
Comment (bombasticSlacks) [inline data/json/items/melee/spears_and_polearms.json:550]: This shouldn't be as good as the properly forged naginata. It currently does *more* dps. I think it wouldn't be unreasonable to have it, have worse balance, it is just made of two pipes and some ductape.

Comment (ANickelN) [reply data/json/items/melee/spears_and_polearms.json:550]: I still think it should be neutral balance, but you're right that the DPS shouldn't be higher than the "real" one.  I took it down a few points.

**Pull request merged**